### PR TITLE
Install manifpy from cmake in CI

### PR DESCRIPTION
**Pull request by GiulioRomualdi**

### CODE REVIEW

**Review by traversaro — APPROVED** [review]

**Comment by GiulioRomualdi** [comment]
Manifpy tries to install the python bindings in /usr

**Comment by traversaro** [comment]
> Manifpy tries to install the python bindings in /usr

You can set manually the `MANIF_PKGDIR`, see https://github.com/robotology/robotology-superbuild/blob/e131566d6f8cdd74b25ed9c5151567f8163cfaec/cmake/Buildmanif.cmake#L23 .

**Review by S-Dafarra — APPROVED** [review]

**Comment by GiulioRomualdi** [comment]
I need to switch to our manif https://github.com/robotology-dependencies/manif.git repo to fix the version

**Comment by GiulioRomualdi** [comment]
iDynTree fails with the following error (perhaps it is an old version and it should be updated @traversaro )

```
Cloning into 'iDynTree'...
HEAD is now at a7fdef001 Release iDynTree 8.0.0 (#1039)
-- The C compiler identification is AppleClang 14.0.0.14000029
-- The CXX compiler identification is AppleClang 14.0.0.14000029
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found LibXml2: /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/lib/libxml2.tbd (found version "2.9.4") 
-- Found YARP: /Users/runner/work/bipedal-locomotion-framework/bipedal-locomotion-framework/install/deps/lib/cmake/YARP (found version "3.8.0")
-- Found YCM: /Users/runner/work/bipedal-locomotion-framework/bipedal-locomotion-framework/install/deps/share/cmake/YCM (found version "0.15.3")
-- Found YARP: /Users/runner/work/bipedal-locomotion-framework/bipedal-locomotion-framework/install/deps/lib/cmake/YARP (found version "3.8.0")
-- Found IPOPT: /usr/local/lib/libipopt.dylib  
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- Found SWIG: /usr/local/bin/swig (found version "4.2.1")  
-- Found Python3: /usr/local/Frameworks/Python.framework/Versions/3.12/bin/python3.12 (found version "3.12.2") found components: Interpreter Development.Module NumPy 
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'distutils'
CMake Error at bindings/CMakeLists.txt:54 (string):
  string sub-command STRIP requires two arguments.
```

**Comment by traversaro** [comment]
> iDynTree fails with the following error (perhaps it is an old version and it should be updated @traversaro )

Yes, that is true. At the superbuild level I fixed this as documented in https://github.com/robotology/robotology-superbuild/issues/1238, i.e. not using `distutils` . However, `distutils` is still use by default, so it probably make sense to cleanup that.

**Comment by traversaro** [comment]
> > iDynTree fails with the following error (perhaps it is an old version and it should be updated @traversaro )
> 
> Yes, that is true. At the superbuild level I fixed this as documented in [robotology/robotology-superbuild#1238](https://github.com/robotology/robotology-superbuild/issues/1238) and [robotology/robotology-superbuild#1508](https://github.com/robotology/robotology-superbuild/pull/1508) , i.e. not using `distutils` . However, `distutils` is still use by default, so it probably make sense to cleanup that.

TL;DR: We need to propagate https://github.com/robotology/robotology-superbuild/pull/1511 to idyntree.

**Comment by GiulioRomualdi** [comment]
Associated PR in iDynTree: https://github.com/robotology/idyntree/pull/1168

**Comment by GiulioRomualdi** [comment]
The ubuntu test fails with
```
/home/runner/work/bipedal-locomotion-framework/bipedal-locomotion-framework/src/IK/tests/QPInverseKinematicsTest.cpp:750: FAILED:
  REQUIRE( error.coeffs().isZero(tolerance) )
with expansion:
  false
```

I was wondering if something similar to what happened back in time on the iDynTree generation of the model. I updated the version of iDynTree in the CI

**Comment by GiulioRomualdi** [comment]
I did a bit of analysis. I was able to replicate the failure https://github.com/ami-iit/bipedal-locomotion-framework/pull/822#issuecomment-2021108836 on my PC with the following commit of iDynTree https://github.com/robotology/idyntree/commit/7e029b963d44fa2606bd608019ba9b82a4398741 while with https://github.com/robotology/idyntree/commit/7ecf7b536be9e03fe05a0d49c8e7adbc5f84dc31 the test pass.


I tried a git bisect and I found the commit in iDynTree that breaks the test in blf https://github.com/robotology/idyntree/commit/9c7f2b36a2505f132700744bfffb142728639a57
```
9c7f2b36a2505f132700744bfffb142728639a57 is the first bad commit
commit 9c7f2b36a2505f132700744bfffb142728639a57
Author: Silvio Traversaro <silvio.traversaro@iit.it>
Date:   Wed Mar 22 14:46:19 2023 +0100

    Add support for PrismaticJoint in getRandomModel

 src/model/include/iDynTree/Model/ModelTestUtils.h | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)
```

**Comment by GiulioRomualdi** [comment]
Our inverse dynamics doesn't support prismatic joints. It would be nice if it could be possible to generate chains with only revolutes joints in iDynTree

**Comment by traversaro** [comment]
See the following related issues:
* https://github.com/robotology/idyntree/pull/1057
* https://github.com/ami-iit/bipedal-locomotion-framework/issues/799
* https://github.com/robotology/idyntree/issues/1149

**Comment by traversaro** [comment]
> Our inverse dynamics doesn't support prismatic joints. It would be nice if it could be possible to generate chains with only revolutes joints in iDynTree

There is the third argument of the function that does that: https://github.com/robotology/idyntree/issues/1149 .

**Comment by GiulioRomualdi** [comment]
Associated fix in iDynTree for generating a model (and not a chain) with revolute joints https://github.com/robotology/idyntree/pull/1171

**Comment by traversaro** [comment]
It seems we need to fix something on the pybind11 bindings, I will check tomorrow.

**Comment by GiulioRomualdi** [comment]
https://github.com/ami-iit/bipedal-locomotion-framework/pull/822#issuecomment-2028477240 probably related to https://github.com/ami-iit/bipedal-locomotion-framework/issues/829

**Comment by GiulioRomualdi** [comment]
I've just rebased on master. Let's see if the CI is happy

**Comment by GiulioRomualdi** [comment]
macOS fails while in solving the qp of the QPInverseKinematics 😭

**Comment by traversaro** [comment]
> macOS fails while in solving the qp of the QPInverseKinematics 😭

You know my position on this. :D:

**Comment by traversaro** [comment]
I guess GitHub has some problems, the CI failures are due to this.

**Comment by GiulioRomualdi** [comment]
Running again the CI
